src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -976,93 +976,97 @@ public static void WriteObject(ref Utf8JsonWriter writer, string key, object obj
 
             Type objType = obj.GetType();
 
-            if (obj is string str)
-                writer.WriteString(key, str);
-            else if (obj is long l)
-                writer.WriteNumber(key, l);
-            else if (obj is int i)
-                writer.WriteNumber(key, i);
-            else if (obj is bool b)
-                writer.WriteBoolean(key, b);
-            else if (obj is DateTime dt)
-                writer.WriteString(key, dt.ToUniversalTime().ToString("O", CultureInfo.InvariantCulture));
-            else if (typeof(IDictionary).IsAssignableFrom(objType))
+            switch (obj)
```

Comment:
This change results in a difference between WriteObjectValue and WriteObject.

Reply:
am reverting to original state